Changing signature of cubic spline resampling cython routine (_csplin…

alexis-roche opened the pull request:
…e_resample3d) for consistency with corresponding _cspline_sample routines. Casting of resampled arrays is now done in python.

jbpoline committed:
NF: allow to pass kwargs to resample

jbpoline committed:
NF: allow ndimage kwarg to be passed to resample

matthew-brett committed:
MAINT: update six to 1.9

Update from hg sources of six.

matthew-brett committed:
MAINT: update pkg_info from dipy copy

Add nipy version printout.  Python 3 compatibility.

matthew-brett committed:
MAINT: update configobj from version 5.0.6

Update configobj, validate from current pypi version (5.0.6).

Edit to isolate from configobj local files.

matthew-brett committed:
MAINT: update testing classes for Python 3

Update nose testing classes for Python 3 compatibility.

Use noseclasses file from numpy master for latest Python 3 fixes.

matthew-brett committed:
RF: remove 2to3 in setup.py install

Remove 2to3 preparing for common codebase.

matthew-brett committed:
RF: use string_types instead of basestring

Python 3 compatibility.

matthew-brett committed:
RF: make relative imports explicit

Explicit relative imports for Python 3 compatibility.

matthew-brett committed:
RF: use modern Exception syntax

Use `except Something as e` and `raise Something('message')`.

matthew-brett committed:
RF: replace uses of backtick for repr

Use %r repr string formatting instead of backticks (backticks gone in
Python 3).

matthew-brett committed:
RF: lists from Python 3 iterators

Use list round functions producing iterators in Pyton 3 such as:

* range
* zip
* dict.keys
* dict.values
* dict.items

Where useful, add range iterators using itertools.chain.

FIXME

matthew-brett committed:
RF: explicit tuple syntax in list comprehension

Specify tuple explicitly with parens for Python 3 compatibility.

matthew-brett committed:
RF: use print_function for print

For Python 3 compatibility.

matthew-brett committed:
RF: use io to import StringIO

Use io module for StringIO import.

matthew-brett committed:
RF: use Python 2 / Python 3 compat iterators

Use Python 3 idioms for iterators, make compatible with Python 2 using
six.

matthew-brett committed:
RF: replace map and filter with list comprehensions

map and filter produce iterable things in Python 3.  Replace with list
comprehensions to give lists in Python 3 and Python 2.

matthew-brett committed:
RF: remove copy of combinations iterator

Combinations iterator present in Python 2.6

matthew-brett committed:
RF: Python 3 dictionary compatibility

* replace `has_key` with `key in dict`;
* replace dict.iteritems with dict.items;
* replace dict.itervalues with dict.values;

matthew-brett committed:
RF: get reduce from function functools

Reduce no longer in global namespace for Python 3.

matthew-brett committed:
RF: remove stub SPM model file

File appears to be some kind of draft, doing nothing.

matthew-brett committed:
RF: change xrange to range

Use range for compatibility with Python 3.

matthew-brett committed:
RF: delete implemention of itertools.permutations

Present in Python 2.6 and above.

matthew-brett committed:
RF: turn off doctests for configobj copy

Turn off doctests as they error on standard test run.

matthew-brett committed:
NF: add utilities module

Start a code support module, ``utilities`.

Add functions:

* is_iterable
* is_numlike

matthew-brett committed:
RF: use utilities instead of operator functions

Use utility module instead of functions from operator module that are
not present in Python 3.

matthew-brett committed:
RF: prefer unicode strings

Use six to specify unicode string where data should be string rather
than bytes.

matthew-brett committed:
TST: add example GLM from doctests to tests

Move example from doctests into unit (regression) test.

Make the test into a literal rst block instead.

matthew-brett committed:
API+RF: remove doctest skipping machinery

The doctest skipping machinery was causing testing failure with the
six.py module, because the machinery runs `getattr` on the object it is
testing which in the case of six.py, was causing various `moves` to be
imported.

As we were only using the doctest skipping machinery for one doctest,
and we've just removed that doctest, remove the machinery.

matthew-brett committed:
BF: fix bug in writing PCA time-courses

We were using the axis number of the PCA analysis (usually axis index 3,
time) as the number of components (by default, 10).  Fix.

I've tested this at the command line with matplotlib, but it's hard to
test automatically without doing some image comparison test, and that
seems like a lot of work.

matthew-brett committed:
DOC: whitespace and docstring cleanup

Whitespace and docstring cleanup for resample, interplation edits.

matthew-brett committed:
BF+TST: fix arg pass to resampling, extend tests

Fix bug where arguments were not being passed to interpolation routine
from `resample_img2img`.

Extend tests and remove commented tests.

Add tests for interpolator

matthew-brett committed:
RF: refactor Transform to allow different inits

Transform object accidentally enforced `__init__` signature, but using
``self.__class__``.  Use ``self.apply`` and return object of class
``Transform`` so we can have subclasses with different ``__init__``
signatures.

matthew-brett committed:
BF+TST: fix resample of not-affine Transforms

Not-affine transforms were not being rounded or thresholded correctly
(as was the case for the Cython routine doing resampling for affine
case).

Not-affine, not default order case was not being reshaped correctly.

matthew-brett committed:
RF+TST: refactor passing extra kwargs to ndimage

We were using a dictionary to pass extra kwargs to ndimage. Instead,
define the relevant arguments explicitly.

alexis-roche committed:
Changing signature of cubic spline resampling cython routine (_cspline_resample3d) for consistency with corresponding _cspline_sample routines. Casting of resampled arrays is now done in python.

alexis-roche committed:
avoid wrapping of large resampled image intensities around zero

alexis-roche committed:
remove cython warnings in cubic spline interpolation module + re-generate c file

alexis-roche commented:
@matthew-brett Are you ok with the suggested changes?

matthew-brett commented on nipy/algorithms/registration/resample.py:
```diff
@@ -16,6 +17,20 @@
+        return np.clip(np.round(arr), mn, mx).astype(dtype)
+    else:
+        return arr.astype(dtype)
+
```
Extra blank line for PEP8

matthew-brett commented:
It would be good to have some basic tests for cast_array .

matthew-brett committed:
Merge pull request #356 from matthew-brett/python3-codebase

MRG: refactor to common Python 2 / 3 codebase

Remove use of 2to3 and update to Python 3 idioms.

Remove some code for Python 2.5 compatibility.
Update code copies from latest upstream.
Remove doctest skipping machinery - a bit complicated for not much gain.

matthew-brett committed:
Merge pull request #355 from matthew-brett/fix-screen-ncomp

MRG: fix bug in writing PCA time-courses

Fix a bug diagnostics output where we were using the axis number as the number of written components.

alexis-roche committed:
add test for cast_array

matthew-brett committed:
BF: fix imports for Python 3

Change to relative imports in numpy fixes file.

matthew-brett committed:
BF: finish fixing local imports

Some remaining local imports in nosetester.py

matthew-brett committed:
BF: wrap more range expressions in list

Some more range expressions that need to be lists (Python 3
compatibility).

matthew-brett committed:
BF: fix for unicode filename test

Probably using the wrong function for the conversion of a Python 2
string to unicode.

matthew-brett committed:
BF: fix coverage package name

Accidentally still using dipy as coverage package name.

matthew-brett commented:
I'll merge this, do the changes.

matthew-brett commented:
Oops, sorry, I didn't notice you'd done the changes already.   I've rebased your changes on top of my branch now, thanks for that.
